### PR TITLE
Add VRT test for validating that split transaction rules work as expected

### PR DESCRIPTION
**Pull request by Marethyu1**
👋 just been experimenting with split rules functionality and added an integration test for it 

The pr contributes to functionality in #1168 that is being discussed in #2379

### CODE REVIEW

**Comment by joel-jeremy** [comment]
Hi @Marethyu1! Is this still in progress? Just a reminder, please don't forget to remove the WIP label if it's ready for review. Thanks!

**Comment by Marethyu1** [comment]
Hey @joel-jeremy,  sorry im not sure if I have permission to remove the label is this something you can help with? Its ready for review :)

**Comment by Marethyu1** [comment]
Ah im not familiar with Traffico, hadn't realised it updated my pr. Wonder If its worth explaining that in the contributing docs?

File: packages/desktop-client/e2e/page-models/account-page.js [hunk 1]
```diff
@@ -141,6 +141,18 @@ export class AccountPage {
   }
 
   async _fillTransactionFields(transactionRow, transaction) {
+    if (transaction.debit) {
```
Comment (joel-jeremy) [inline packages/desktop-client/e2e/page-models/account-page.js:144]: Question: Why was this moved up?

File: packages/desktop-client/src/components/modals/EditRule.jsx [hunk 2]
```diff
@@ -1085,6 +1085,7 @@ export function EditRule({ modalProps, defaultRule, onSave: originalOnSave }) {
                     onClick={() => {
                       addActionToSplitAfterIndex(actionSplits.length, -1);
                     }}
+                    data-testid="split-transactions-add"
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/modals/EditRule.jsx:1088]: Nit: 
```suggestion
                    data-testid="add-split-transactions"
```

File: packages/desktop-client/e2e/page-models/rules-page.js [hunk 3]
```diff
@@ -61,6 +61,29 @@ export class RulesPage {
...
+        idx = data.splits.beforeSplitActions.length;
+        await this._fillEditorFields(
+          data.splits.beforeSplitActions,
+          this.page.getByTestId('action-list'),
+        );
+      }
+
+      if (data.splits.splitActions) {
+        for (const splitActions of data.splits.splitActions) {
+          await this.page.getByTestId('split-transactions-add').click();
```
Comment (joel-jeremy) [inline packages/desktop-client/e2e/page-models/rules-page.js:78]:
```suggestion
          await this.page.getByTestId('add-split-transactions').click();
```

File: packages/desktop-client/e2e/page-models/rules-page.js [hunk 4]
```diff
@@ -61,6 +61,29 @@ export class RulesPage {
         this.page.getByTestId('action-list'),
       );
     }
+
+    if (data.splits) {
+      let idx = 0;
+
+      if (data.splits.beforeSplitActions) {
+        idx = data.splits.beforeSplitActions.length;
```
Comment (joel-jeremy) [inline packages/desktop-client/e2e/page-models/rules-page.js:69]: We can just put this inside the splitActions if block below

**Review by joel-jeremy — CHANGES_REQUESTED** [review]

File: packages/desktop-client/e2e/page-models/rules-page.js (hunk 4 above)
Comment (Marethyu1) [reply packages/desktop-client/e2e/page-models/rules-page.js]: Done 👍
